src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -243,8 +243,8 @@ public long getValidationTimeout()
    @Override
    public void setValidationTimeout(long validationTimeoutMs)
    {
-      if (validationTimeoutMs < 250) {
-         throw new IllegalArgumentException("validationTimeout cannot be less than 250ms");
+      if (validationTimeoutMs < 1000) {
+         throw new IllegalArgumentException("validationTimeout cannot be less than 1000ms");
       }
```

Comment:
That is incorrect, you need to read the entire thread.  Having a ``validationTimeout`` that is higher than the ``connectionTimeout`` leads to violations of the ``connectionTimeout`` contract.  The ``violationTimeout`` absolutely must be able to be set to an equal or lesser value than ``connectionTimeout``.

I am going to cherry-pick the changes in this pull request rather than discussing back and forth on every change.